### DISCUSSION TITLE
cannot open source file "netinet/in.h" (dependency of "C:\vcpkg\packages\aws-sdk-cpp_x86-windows\include\aws\kinesis\KinesisClient.h")

### DISCUSSION THREAD

**Discussion by SankarshanRoy** [aws/aws-sdk-cpp#2400] [Q&A] [answered]
I am using Platformio , i download aws-sdk-cpp using vcpkg , only the kinesis . I am using a Pi pico w. 

platformio.ini code:

[env:rpipicow]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = rpipicow
framework = arduino
board_build.core = earlephilhower
board_build.filesystem_size = 0.5m
build_flags =
    -I"C:/vcpkg/packages/aws-sdk-cpp_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-auth_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-cal_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-common_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-compression_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-event-stream_x86-windows/include"
    -I"C:/vcpkg/packages/aws-checksums_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-http_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-io_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-mqtt_x86-windows/include"
    -I"C:/vcpkg/packages/aws-crt-cpp_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-s3_x86-windows/include"
    -I"C:/vcpkg/packages/aws-c-sdkutils_x86-windows/include"
lib_deps =
    aws-sdk-cpp_x86-windows_Kinesis

upload_port = D:
monitor_speed = 115200 
monitor_port = COM5

I am getting ERROR:
![image](https://user-images.githubusercontent.com/128494997/226745596-642b9b1e-d6c7-41a9-ad04-d0a9274ea2af.png)
PLEASE HELP
!

**Comment by jmklix** [comment]
### Building aws-sdk-cpp - Using vcpkg

You can download and install aws-sdk-cpp using the [vcpkg](https://github.com/Microsoft/vcpkg) dependency manager:

    git clone https://github.com/Microsoft/vcpkg.git
    cd vcpkg
    ./bootstrap-vcpkg.sh
    ./vcpkg integrate install
    ./vcpkg install aws-sdk-cpp

The aws-sdk-cpp port in vcpkg is kept up to date by Microsoft team members and community contributors. If the version is out of date, please [create an issue or pull request](https://github.com/Microsoft/vcpkg) on the vcpkg repository.

**Reply by SankarshanRoy** [reply]
I did that,  and i used ./vcpkg install aws-sdk-cpp[Kinesis] , the library was installed in the path mentioned in the ini file

**Reply by jmklix** [reply, marked as answer]
This doesn't look like it's a problem with the sdk. Please check if the problem can be fixed with visual studio settings: https://stackoverflow.com/questions/12748896/cannot-open-windows-h-in-microsoft-visual-studio
